scripts/BuildAndTest.ps1

```diff
@@ -214,7 +222,10 @@ if (-not $?) {
 if (-not $NoBuild) {
     Invoke-DotNetBuild $SolutionFile
     if ($ENV:OS) {
-        Invoke-DotNetBuild $sampleSolutionFile
+        if (-not $NoSamples) {
```

Comment:
Same idea as above, any reason these couldn't be combined into a single if statement?